libsoftwarecontainer/src/softwarecontainer.cpp

```diff
@@ -197,6 +197,19 @@ ReturnCode SoftwareContainer::startGateways(const GatewayConfiguration &gwConfig
 
 ReturnCode SoftwareContainer::configureGateways(const GatewayConfiguration &gwConfig)
 {
+    assertValidState();
+
+    // Make sure that all the gateway ids in the given GatewayConfig can map
+    // to a gateway in SoftwareContainer
+    for (auto &id : gwConfig.ids()) {
+        for (auto &gateway : m_gateways) {
+            if (id != m_gateways) {
```

Comment:
how come you compare string id with vector of  Gateway pointer
is this really compiles ?